dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/adapter/OutputFormatter.java

```diff
@@ -0,0 +1,127 @@
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.common.adapter;
+
+import java.math.BigDecimal;
+
+/**
+ * Simple component responsible do enforce specific output/format to types where
+ * this is required.
+ *
+ * Its goal is to provide methods that can improve the String representation of
+ * particular types. Should be used specially for serialization purposes and in
+ * very specific cases.
+ *
+ * @author maikel arabori
+ */
+class OutputFormatter
+{
+    private static final int TEN_MILLION = 10000000;
+
+    private OutputFormatter()
+    {
+    }
+
+    /**
+     * This method should support different formatters and apply them based on
+     * the input object type.
+     *
+     * Currently only Double type is supported.
+     *
+     * @param object represents the object to be formatted.
+     *
+     * @return a String containing the full value of the number object. If the
+     *         type is not supported or is null it will return the given
+     *         parameter object itself
+     */
+    static Object maybeFormat( final Object object )
+    {
+        if ( object instanceof Double )
+        {
+            return formatDouble( (Double) object );
+        }
+
+        return object;
+    }
+
+    /**
+     * This method will simply format a Double object to a non-scientific
+     * notation. It will happen only if the given argument is a Double.
+     *
+     * By default, depending on the type, Java will convert number types, that
+     * are equals or greater than 10 million, to scientific notation when
+     * transformed to String.
+     *
+     * For example, the Double 99999999 becomes 9.9999999E7 (scientific) when
+     * printed.
+     *
+     * So, this method aims to avoid such representations and instead returns
+     * the full value as String (in this example it would be "99999999.0").
+     *
+     * @param doubleValue represents the object that should be a Double
+     *
+     * @return a String containing the full value of the number object. If the
+     *         type is not supported or is null it will return the given
+     *         parameter object itself
+     *
+     * @throws NullPointerException if doubleValue is null
+     */
+    private static Object formatDouble( final Double doubleValue )
+    {
+        // Don't waste resources if the value is smaller than 10
+        // million as the default representation will be the expected one.
+        if ( doubleValue >= TEN_MILLION )
+        {
+            // We need to pass a String to the constructor, otherwise we
+            // loose precision.
+            final String value = new BigDecimal( (doubleValue).toString() ).toPlainString();
+            final boolean hasDecimalDigit = value.contains( "." );
+
+            // Here, we ensure a ".0" at the end when there is no decimal
```

Comment:
Could we move this logic to a separate method, and use standard method Javadocs to avoid large in-line comments?

Reply:
Changes applied @larshelge